.github/workflows/build-deploy.yml

```diff
@@ -0,0 +1,123 @@
+on:
```

Comment:
I had intended to call the deploy workflow from PIE, however after some investigation, the logic is a bit too specific and would require some refactoring.

For now I've copied the bulk of the logic, and removed the bits that aren't needed for this repo.

Perhaps in the future, we can look to move reusable workflows into their own repo to make them completely repo agnostic